### PR TITLE
List page sort implementation

### PR DESCRIPTION
**Pull request by y7chiu**
Me and Alex are making:
- The function of sorting the list by name (Alphabetically) and timestamp.
- Make the function of making arrow up/down when sorting.
- make the title name, tag, and timestamp center to its corresponding category.
- Delete the "tag:" and "timestamp:" in list entry and change the name for the timestamp to Date Created in category bar.

*Notes: I think the searching for title feature has the problem in our version. We could not search for the title to get the list entry, however, all the data is in the localStorage. As we tried the search feature in mvp version, it has the same problem as well. Only this problem occurs in the list-page-sort branch. I did not see the problem what Warren said. It can make the title for different entries and will not cause problem.

### CODE REVIEW

File: src/scripts/list.js
Comment (SatvikS7) [inline src/scripts/list.js:1]: Please add javadoc comments on function sortByCategory for JSDoc action

**Review by SatvikS7 — CHANGES_REQUESTED** [review]
Changes look good and function as described. Adding the sort is a great feature. Please look at comment on list.js for change request and add some comments to list.css.

File: src/scripts/list.js
Comment (y7chiu) [reply src/scripts/list.js:1]: Added the javadoc comments for function sortByCategory and updateSortArrows. Right now, still need to fix the merge conflict.

**Review by SatvikS7 — CHANGES_REQUESTED** [review]
Very small fix but make sure comments in javascript files are done in javadoc format. Ex (taken from list.js lines  272-279)

/**
 * Searches all journal entries for a string only if the entries include all the specified tags and is within the time period filter.
 * @param {string} query - exact string to search for
 * @param {string} query - exact string to search for 
 * @param {Array.string} tags - list of exact tags to include
 * @param {string} startDate - start date formatted yyyy-mm-dd
 * @param {string} endDate - end date formatted yyyy-mm-dd
 * @returns {JournalEntry[]} matching entries
 * @returns matching entries
 */

Basically, it should contain description of the function, parameters, and their types and what they represent, and the output description.

**Review by SatvikS7 — APPROVED** [review]
Ignore previous comment, documentation and function looks good, merging to list-page branch
